x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/cert/CertificateInfo.java

```diff
@@ -46,7 +46,7 @@ public CertificateInfo(String path, String format, String alias, boolean hasPriv
         this.path = path;
         this.format = Objects.requireNonNull(format, "Certificate format cannot be null");
         this.alias = alias;
-        this.subjectDn = Objects.requireNonNull(certificate.getSubjectDN().getName());
+        this.subjectDn = Objects.requireNonNull(certificate.getSubjectX500Principal().getName());
```

Comment:
As commented above, we can retain the existing behaviour but still move forward with the main purpose of this PR by using `certificate.getSubjectX500Principal().toString()`. If we go ahead with this suggestion, I'd also add a comment above this line to explain the reasoning.